src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineTransportFactory.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine;
+
+use Symfony\Bridge\Doctrine\RegistryInterface;
+use Symfony\Component\Messenger\Transport\Serialization\Serializer;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class DoctrineTransportFactory implements TransportFactoryInterface
+{
+    private $registry;
+    private $serializer;
+    private $debug;
+
+    public function __construct(RegistryInterface $registry, SerializerInterface $serializer = null, $debug = false)
+    {
+        $this->registry = $registry;
+        $this->serializer = $serializer ?? Serializer::create();
+        $this->debug = $debug;
+    }
+
+    public function createTransport(string $dsn, array $options): TransportInterface
+    {
+        $manager = str_replace('doctrine://', '', $dsn);
+
+        try {
+            $connection = new Connection($options['table_name'] ?? 'messenger_messages', $this->registry->getConnection($manager), $this->debug);
+        } catch (\InvalidArgumentException $e) {
+            throw new \RuntimeException(
+                sprintf('Unable to create Doctrine transport for Messenger (dsn "%s"): %s', $dsn, $e->getMessage()),
```

Comment:
I wouldn't expose the dsn in the exception's message, this is an obvious breach of security. Actually, I would create a proper exception (and catch the proper exception too...) and throw it back, and give the dsn as a context (not part of the message).

Reply:
Actually the DSN will look like `doctrine://<entity_manager>` 
There will no be database credentials in there

Reply:
This is still leaking some information on the message that should belong to a context rather than the exception message IMO

Reply:
@Taluu i've made changes regarding your comment